routes/users.js

```diff
@@ -0,0 +1,112 @@
+import express from 'express';
+import bcrypt from 'bcryptjs';
+
+import { addUserValidationRules, validateData } from '../validate.js';
+import { User } from '../models';
+
+const router = express.Router();
+
+const getUsers = async (request, response) => {
+  try {
+    const users = await User.findAll();
+    return response.status(200).send(users);
+  } catch (error) {
+    return response.status(400).send('An error occured.');
+  }
+};
+
+const addUser = async (request, response) => {
+  const { userName, email, firstName, lastName, password } = request.body;
+
+  try {
+    const foundUser = await User.findAll({
+      where: {
+        userName,
+      },
+    });
+
+    if (foundUser.length !== 0) {
+      return response.status(302).json({
```

Comment:
`3xx` response codes are for redirections. Here you could use `422 Unprocessable Entity` or a `403 Conflict`.